src/hooks/useLoggedInUser.tsx

```diff
@@ -8,11 +8,20 @@ const UserContext = createContext<User | undefined>(undefined);
 
 export const UserProvider: FC<PropsWithChildren> = ({ children }) => {
 	// Hold user info in state
-	const [user, setUser] = useState<User>();
+	const [user, setUser] = useState<User>(
+		localStorage.getItem('user')
+			? JSON.parse(localStorage.getItem('user'))
+			: undefined
+	);
 
 	// Setup onAuthChanged once when component is mounted
 	useEffect(() => {
-		onAuthChanged(u => setUser(u ?? undefined));
+		onAuthChanged(u => {
+			setUser(u ?? undefined);
```

Comment:
A tady taky?

```
Argument of type 'User | undefined' is not assignable to parameter of type 'SetStateAction<User>'.
  Type 'undefined' is not assignable to type 'SetStateAction<User>'
```

Ani jedno mi nedává smysl. :D

Reply:
Byl špatně zadefinovaný typ stavu:
`useState<User>` namísto `useState<User | undefined>`

Reply:
Předtím se tam ale taky nastavoval `(u ?? undefined)` a typ u `useState` byl pouze `<User>`. Nicméně to předtím neřvalo.

Reply:
Asi chyba kompilátoru, že to neodhalil :D